### PR TITLE
Add invariant checks to fuzz tests

### PR DESCRIPTION
**Pull request by sentilesdal**
Adds the following checks:

 1. PV always >= idle
 2. ∆ lp share price should never be more than 0.1%

Part of #1123

### CODE REVIEW

File: lib/agent0/bin/fuzz_bot_invariant_checks.py [hunk 1]
```diff
@@ -231,6 +247,84 @@ def run_invariant_checks(
         raise FuzzAssertionException(*exception_message, exception_data=exception_data)
 
 
+def _lp_share_price(
+    block_number: BlockNumber,
+    interface: HyperdriveReadInterface,
+    test_epsilon: float,
+    pool_state: PoolState,
+    exception_message: list[str],
+    exception_data: dict[str, Any],
```
Comment (slundqui) [inline lib/agent0/bin/fuzz_bot_invariant_checks.py:256]: Are you sure these are passed in via reference? Can we return the exception message and exception data just to make it explicit that these two fields are getting adjusted? Better yet, don't pass these in and add the message/data in the caller

File: lib/ethpy/ethpy/hyperdrive/interface/read_interface.py [hunk 2]
```diff
@@ -320,6 +320,29 @@ def get_vault_shares(self, block_number: BlockNumber | None) -> FixedPoint:
             block_number = self.get_block_number(self.get_current_block())
         return _get_vault_shares(self.yield_contract, self.hyperdrive_contract, block_number)
 
+    def get_idle_shares(self, block_number: BlockNumber | None) -> FixedPoint:
```
Comment (slundqui) [inline lib/ethpy/ethpy/hyperdrive/interface/read_interface.py:323]: You'll want to double check with @dpaiton here, but I think the read interface might be strictly contract calls without math on our end

Comment (dpaiton) [reply lib/ethpy/ethpy/hyperdrive/interface/read_interface.py:323]: We don't have a rust function for this yet, so I think it's ok to do it in python. However it's not using an RPC to get the balance so I think the docstring is incorrect.

Comment (sentilesdal) [reply lib/ethpy/ethpy/hyperdrive/interface/read_interface.py:323]: Added an issue to hyperdrive-sdk: https://github.com/delvtech/hyperdrive-sdk/issues/55

File: lib/agent0/bin/fuzz_bot_invariant_checks.py (hunk 1 above)
Comment (sentilesdal) [reply lib/agent0/bin/fuzz_bot_invariant_checks.py:256]: yeah lists and dicts are always passed by reference but i like what you're saying about making it more explicit anyway.

**Review by dpaiton — APPROVED** [review]
awesome!

**Pull request merged**